HIP-XXXX.md

```diff
@@ -0,0 +1,298 @@
+
+
+## Timing considerations
+
+Certificates should include tree roots/proofs that overlaps the intervals for synced and out-of-sync (1-week behind) clients. In most cases, only a single proof is required. If there were UPDATEs within last 32 urkel intervals, two proofs may be required refer to servers section for more details.
+
+
+```
+------------------------------ Urkel intervals --------------------------------->
+                                         [              synced client           ]
+        [              out-of-sync client           ][          1-week          ]
+                                                .<--------- x509 cert  ----------
+                                                ^
+                                                Included root/proof
+                                                
+```
+
+Illustrating included proofs (urkel extension 1.2/1.3 sections) with modified DS/HIP-5 delegation 
+the certificate includes two proofs. The recent proof applies to synced clients.
+
+```
+------------------------------ Urkel intervals --------------------------------->
+                                         [              synced client           ]
+        [              out-of-sync client           ][          1-week          ]
+                                                .<-------------------------------
+                                                ^
+                                                old root/proof
+                                                               .<----------------
+                                                               ^
+                                                           new root/proof
+                                                
+```
+
+
+## Censorship resistance
+
+While still being completely trustless, the proposed client relies on forwarding resolvers. These resolvers can censor DNS queries. You may think that a natural solution is falling back to recursion if the client detects censorship, but recursive resolvers today are easily censorable, so it's unlikely to help. Since forwarding resolvers do not need to be trusted, clients may rely on resolver aggregators & anonymized DNS relays, which are more lightweight, private, and have better censorship resistance properties. Future standards may define such techniques.
+
+
+## Security considerations
+
+Clients should dowload block headers & verify proof of work as defined in the theory of operation. It may also be possible to rely on Scalable Transparent Argument of Knowledge (STARK) to generate small proofs that clients can verify. Future standards may define how these proofs are generated & shared with clients to reduce state and reliance on full nodes. The tree root is the main trust anchor for securing HTTPS so great care must be taken to retrieve it securely. This standard does not restrict how the tree roots are obtained as long as they're deemed secure by the client.

```

Comment:
It may also be possible in the future to add (like) 12 HNS block headers to the x509 extension once that is expensive enough PoW-wise, and the client won't need any hns p2p machinery at all. These headers could also be cached and re-used.

Reply:
added thanks!